cache.go

```diff
@@ -146,12 +139,17 @@ func (a *Cache[K, V]) Set(k K, v V) {
 }
 
 // Replaces existing values, which are evicted.
-// A min size of 1 will be used.
+// A min size of 1 will be used. Set item always comes out of evict.
```

Comment:
Clarifying we always evict after a SetS, even if immediate. In the immediate case added here that ensures the counting cache's SetS returned Handle does the final Node release.